### PR TITLE
KIN-79: Created toggle service to fetch features.

### PR DESCRIPTION
**Pull request by deandalm**
#### Purpose
- [X] Feature
- [ ] Bug Fix
- [ ] Hotfix
- [ ] Other

#### Associated Tickets
- [Feature service](https://silvercar.atlassian.net/browse/KIN-79)

#### Details
- Adding feature service and models 
- Will add tests in a future PR, checking this in so others can build functionality with it.

#### Checklist
- [ ] Tests
- [ ] Correct base and merge branches

### CODE REVIEW

File: unleash/Services/ToggleService.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    func fetchToggles(url: URL) -> Promise<Features>
+}
+
+extension ToggleService {
+    func fetchToggles(url: URL) -> Promise<Features> {
+        let togglesUrl = url.appendingPathComponent("/client/features")
+        return firstly {
+            URLSession.shared.dataTask(.promise, with: try makeUrlRequest(url: togglesUrl)).validate()
+            }.map {
+                return try JSONDecoder().decode(Features.self, from: $0.data)
```
Comment (johnny-1111one1) [inline unleash/Services/ToggleService.swift]: The register service uses `try JSONSerialization.jsonObject(with: $0.data)`, is there a difference using the two?

File: unleash/Services/ToggleService.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        }
+    }
+    
+    private func makeUrlRequest(url: URL) throws -> URLRequest {
+        var request = URLRequest(url: url)
+        request.httpMethod = "GET"
+        request.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        request.addValue("application/json", forHTTPHeaderField: "Accept")
+        request.addValue("UNLEASH-APPNAME", forHTTPHeaderField: self.appName)
```
Comment (johnny-1111one1) [inline unleash/Services/ToggleService.swift]: Are these 3 headers backwards? Should the value be first, then the header?

File: unleash.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -195,19 +216,25 @@
 			);
 			path = Services;
 			sourceTree = "<group>";
-                };
-		9D1C643622DF6AD30069FD45 /* Services */ = {
+		};
+		9D1C645322E003EE0069FD45 /* Recovered References */ = {
```
Comment (johnny-1111one1) [inline unleash.xcodeproj/project.pbxproj]: What does this mean?

Comment (deandalm) [reply unleash.xcodeproj/project.pbxproj]: I don't know, Xcode generated this.

File: unleash/Services/ToggleService.swift (hunk 1 above)
Comment (deandalm) [reply unleash/Services/ToggleService.swift]: I believe this is how you get it to map to a `Codeble` object.

File: unleash/Services/ToggleService.swift (hunk 2 above)
Comment (deandalm) [reply unleash/Services/ToggleService.swift]: Oops, I'll fix that.

**Comment by deandalm** [comment]
@johnny-1111one1 fixed that header issue but the last CircleCI ran had a Danger issue:
```
`fetch_pull_request_url': undefined method `first' for nil:NilClass (NoMethodError)
```
Any idea why?

**Comment by johnny-1111one1** [comment]
@deandalm I'm pretty sure that's because you did a force push.

**Comment by deandalm** [comment]
@johnny-1111one1 interesting, well it worked now. Just need someone to review this.

**Pull request merged**